src/pages/storage/AttachDiskDeviceModal.tsx

```diff
@@ -0,0 +1,99 @@
+import { FC, KeyboardEvent, useState } from "react";
+import { Modal } from "@canonical/react-components";
+import FormLink from "components/FormLink";
+import BackLink from "components/BackLink";
+import { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
+import CustomVolumeModal from "./CustomVolumeModal";
+import { LxdStorageVolume } from "types/storage";
+import { LxdDiskDevice } from "types/device";
+import HostPathDeviceModal from "./HostPathDeviceModal";
+
+type DiskDeviceType = "custom volume" | "host path" | "";
+
+interface Props {
+  close: () => void;
+  onFinish: (device: LxdStorageVolume | LxdDiskDevice) => void;
+  formik: InstanceAndProfileFormikProps;
+  project: string;
+}
+
+const AttachDiskDeviceModal: FC<Props> = ({
+  close,
+  formik,
+  project,
+  onFinish,
+}) => {
+  const [type, setType] = useState<DiskDeviceType>("");
+
+  const handleEscKey = (e: KeyboardEvent<HTMLElement>) => {
+    if (e.key === "Escape") {
+      close();
+    }
+  };
+
+  const handleGoBack = () => {
+    if (type) {
+      setType("");
+      return;
+    }
+  };
+
+  const modalTitle = !type ? (
+    "Choose disk device type"
```

Comment:
```suggestion
    "Choose type"
```
I think the options and the context make it clear this is about disk devices :)

Reply:
On second thought: "Choose disk type" might be better.